chitech_composite_processor/Utils_Info.py

```diff
@@ -0,0 +1,202 @@
+def GenerateSpectrumData(neutron_gs, psi, sigma_heat, gamma_gs=[]):
+  G_neutron = len(neutron_gs)
+  G_gamma   = len(gamma_gs)
+  G         = G_neutron + G_gamma
+  assert len(psi) == G, \
+    "Total neutron+gamma groups not compatible with psi."
+  assert len(sigma_heat) == G, \
+    "Total neutron+gamma groups not compatible with heating XS"
+
+  #==================== Generate neutron spectrum data
+  n_bndrys, n_vals = [], []
+  n_heating = []
+  for g in range(G_neutron):
+    gprime    = G_neutron - g - 1
+    lo_bound  = neutron_gs[g][1]*1.0e-6
+    hi_bound  = neutron_gs[g][2]*1.0e-6
+    bin_width = hi_bound-lo_bound
+    spectrum  = psi[G_neutron-g-1] / bin_width
+    n_bndrys += [lo_bound, hi_bound]
+    n_vals   +=  [spectrum, spectrum]
+
+    heating_spectrum  = sigma_heat[G_neutron-g-1]
+    n_heating += [heating_spectrum, heating_spectrum]
+
+  n_bndrys = np.array(n_bndrys)
+  n_vals   = np.array(n_vals)
+  n_heating = np.array(n_heating)
+
+  #==================== Generate gamma spectrum data
+  g_bndrys, g_vals = [], []
+  g_heating = []
+  for g in range(G_gamma):
+    gprime = (G_gamma - g - 1) + G_neutron
+    lo_bound  = gamma_gs[g][1]*1.0e-6
+    hi_bound  = gamma_gs[g][2]*1.0e-6
+    bin_width = hi_bound-lo_bound
+    spectrum  = psi[gprime] / bin_width
+    g_bndrys += [lo_bound, hi_bound]
+    g_vals   += [spectrum, spectrum]
+
+    heating_spectrum  = sigma_heat[gprime]
+    g_heating += [heating_spectrum, heating_spectrum]
+
+  g_bndrys = np.array(g_bndrys)
+  g_vals   = np.array(g_vals)
+  g_heating = np.array(g_heating)
+
+  return [(n_bndrys, n_vals), (g_bndrys, g_vals), (n_heating, g_heating)]
+
+#====================================================================
+def InfiniteMediumSpectrum(data, path, plot=False):
+  neutron_gs = data["neutron_gs"]
+  gamma_gs = data["gamma_gs"]
+  sig_t = data["sigma_t"]
+  transfer_mats = data["transfer_matrices"]
+  transfer_mats_nonzeros = data["transfer_matrices_sparsity"]
+  sig_heat = data["sigma_heat"]
+
+  #======================================= Get data from dictionary
+  G_neutron = len(neutron_gs)
+  G_gamma   = len(gamma_gs)
+  coupled_txt = ''
+  if G_gamma>0:
+    coupled_txt = '_coupled_ng'
+
+  G = np.size(sig_t)
+  
+  v_sig_t = np.array(sig_t)
+  M_sig_gp_to_g = np.zeros([G,G])
+
+  for gprime in range(G):
+    for g in transfer_mats_nonzeros[0][gprime]:
+      M_sig_gp_to_g[gprime,g] = transfer_mats[0][gprime][g]
+
+  #======================================= Solve for psi
+  S = M_sig_gp_to_g.transpose()
+  T = np.diag(v_sig_t)
+
+  if plot:
+    fig = plt.figure()
+    plt.matshow(np.log(S))
+    if G_gamma>0:
+      g_txt = '_g'+str(G_gamma)
+    else:
+      g_txt = ''
+    filename = path+'transfert_matrix_n'+str(G_neutron)+g_txt+'.png'
+    plt.savefig(filename)
+
+  A = T - S
+  A_inv = np.linalg.inv(A)
+
+  v_src = np.zeros(G)
+  # Get the source term:
+  src_term = create_source_spectrum(neutron_gs, 14.0, fission = False)

```

Comment:
1) create source must be aware of n or g.
2) the original InfiniteMediumSpectrum must be updated with the new create_source_spectrum
3) in addition, the orignal InfiniteMediumSpectrum has all of the plotting "garbage" in it. It must to move out
4) therefore, you do not need a new version of InfiniteMediumSpectrum